streamer/src/streamer.rs

```diff
@@ -36,13 +36,18 @@ fn recv_loop(
     recycler: &PacketsRecycler,
     name: &'static str,
     coalesce_ms: u64,
+    use_pinned_memory: bool,
 ) -> Result<()> {
     let mut recv_count = 0;
     let mut call_count = 0;
     let mut now = Instant::now();
     let mut num_max_received = 0; // Number of times maximum packets were received
     loop {
-        let mut msgs = Packets::new_with_recycler(recycler.clone(), PACKETS_PER_BATCH, name);
+        let mut msgs = if use_pinned_memory {
+            Packets::new_with_recycler(recycler.clone(), PACKETS_PER_BATCH, name)
+        } else {
+            Packets::new_unpinned_with_recycler(recycler.clone(), PACKETS_PER_BATCH, name)
```

Comment:
does the unpinned one need a recycler?

also, I think using the same recylcer for both pinned and unpinned ones will hurt performance when an unpinned vector is returned from the recycler but we need a pinned vector.

I think, the unpinned one just doesn't need a recycler.

Reply:
yea, it doesn't need it.

Reply:
I think in general we shouldn't mix pinned/unpinned recyclers. As far as I know, there wouldn't be any mixing with this PR. A `receiver` is created either with `use_pinned_memory=true` or `=false`, and then for the lifetime of that receiver it would always use either pinned or unpinned.